src/PlusServer/Commands/vtkPlusCapistranoCommand.cxx

```diff
@@ -0,0 +1,321 @@
+      status = PLUS_FAIL;
+      continue;
+    } 
+    vtkPlusCapistranoVideoSource* device = dynamic_cast<vtkPlusCapistranoVideoSource*>(aDevice);
+
+    if (parameterName == vtkPlusCapistranoVideoSource::CAPISTRANO_SET_WOBBLE_RATE)
+    {
+      int wobbleRate_int = std::stoi(value);
+      status = device->SetWobbleRate((unsigned char)wobbleRate_int);
+      resultString += " Success=\"true\"/>";
+      metaData[parameterName] = std::make_pair(IANA_TYPE_US_ASCII, "SUCCESS");
+    }
+    if (parameterName == vtkPlusCapistranoVideoSource::CAPISTRANO_GET_WOBBLE_RATE)
+    {
+      value = std::to_string(device->GetWobbleRate());
+      std::stringstream ss;
+      ss << value;
+      resultString += " Success=\"true\"";
+      resultString += " Value=\"" + ss.str() + "\"";
+      metaData[parameterName] = std::make_pair(IANA_TYPE_US_ASCII, ss.str());
+      status = PLUS_SUCCESS;
+    }
+    if (parameterName == vtkPlusCapistranoVideoSource::CAPISTRANO_SET_BIDIRECTIONAL_MODE)
+    {
+      bool bidi = std::stoi(value);
+      status = device->SetBidirectionalMode(bidi);
+      resultString += " Success=\"true\"/>";
+      metaData[parameterName] = std::make_pair(IANA_TYPE_US_ASCII, "SUCCESS");
+    }
+    if (parameterName == vtkPlusCapistranoVideoSource::CAPISTRANO_FREEZE_PROBE)
+    {
+      bool freezeProbe_bool  = std::stoi(value);
+      status = device->FreezeDevice(freezeProbe_bool);
+      resultString += " Success=\"true\"/>";
+      metaData[parameterName] = std::make_pair(IANA_TYPE_US_ASCII, "SUCCESS");
+    }
+    if (parameterName == vtkPlusCapistranoVideoSource::CAPISTRANO_IS_PROBE_FROZEN)
+    {
+      value = device->IsFrozen() ? "true" : "false";
+      std::stringstream ss;
+      ss << value;
+      resultString += " Success=\"true\"";
+      resultString += " Value=\"" + ss.str() + "\"";
+      metaData[parameterName] = std::make_pair(IANA_TYPE_US_ASCII, ss.str());
+      status = PLUS_SUCCESS;
+    }
+    if (parameterName == vtkPlusCapistranoVideoSource::CAPISTRANO_SET_JITTER_COMPENSATION)
+    {
+      int jitterCompensation_int = std::stoi(value);
+      status = device->SetJitterCompensation((unsigned char)jitterCompensation_int);
+      resultString += " Success=\"true\"/>";
+      metaData[parameterName] = std::make_pair(IANA_TYPE_US_ASCII, "SUCCESS");
+    }
+    if (parameterName == vtkPlusCapistranoVideoSource::CAPISTRANO_GET_JITTER_COMPENSATION)
+    {
+      value = std::to_string(device->GetJitterCompensation());
+      std::stringstream ss;
+      ss << value;
+      resultString += " Success=\"true\"";
+      resultString += " Value=\"" + ss.str() + "\"";
+      metaData[parameterName] = std::make_pair(IANA_TYPE_US_ASCII, ss.str());
+      status = PLUS_SUCCESS;
+    }
+    if (parameterName == vtkPlusCapistranoVideoSource::CAPISTRANO_SET_MIS_MODE)
+    {
+      bool MISMode_bool = std::stoi(value);
+      status = device->SetMISMode(MISMode_bool);
```

Comment:
Though it probably won't happen in practice since we generally build with the correct sdk, this function (and a few others) have a chance of failure (https://github.com/PlusToolkit/PlusLib/blob/62ab3159e91eda69b2b19ffd22e46bde74590997/src/PlusDataCollection/Capistrano/vtkPlusCapistranoVideoSource.cxx#L1471). 

SetUsParameter handles the case of commands failing to actually do the thing they're supposed to by relying on the output of `SetNewImagingParameters`/`InternalApplyImagingParameterChanges`. 

Is there something similar we can do here to ensure that the success/failure status return sent back by the command reflects the actual state of the device post-command-being-sent?

Reply:
Like checking the status of the command and adding a log and plus status change?
(for commands: GetMISMode, SetMISPlusePeriod, FreezeDevice) 
EX: https://github.com/nhjohnston/PlusLib/blob/35e8d8eea6c6faa043c65d6cc2590a74dcca512f/src/PlusDataCollection/Capistrano/vtkPlusCapistranoVideoSource.cxx#L2123-L2132

Reply:
Yes, something like that should work